nemo/collections/nlp/modules/common/text_generation_utils.py

```diff
@@ -748,6 +750,9 @@ def sample_sequence_batch(
 
                 # Clamp the predicted out of vocabulary tokens
                 prev = torch.clamp(prev, max=tokenizer.vocab_size - 1)
+                # force EOS token when tokens_to_generate has been reached for an item in the batch.
+                prev[context_length >= maxlens] = tokenizer.eos_id
```

Comment:
This looks ok to me, but there might be an edge case if we happen to use a text generation strategy that doesn't consider `EOS` as the end of generation (the computed length would be incorrect). Is this something that needs to be supported? If yes, I think two things could be done:

1. Do something like `done_token = done_token | (context_length == maxlens).byte()` after the definition of `done_token` below (to be sure we consider generation to be done even if the generation strategy ignores the `EOS` token)
2. Use `prev[context_length == maxlens] = tokenizer.eos_id` here (no need for `>=` since `is_done` will be set for future steps, which will force `EOS` on l.759)

Reply:
what about using the padding token?

Reply:
> Is this something that needs to be supported?

As discussed, I'm ok not supporting it for now.

> what about using the padding token?

As discussed, there are some additional subtleties in properly handling padded inputs => personally I'm ok with not doing it in this PR.